neaten/neaten.go

```diff
@@ -0,0 +1,390 @@
+		t.Date,
+		filepath.Base(interestBaseDir),
+		filepath.Base(interestUniqueDir),
+		filepath.Base(multiUniqueDir),
+		suffix))
+
+	return t.renameAndMatchPerms(source, dest)
+}
+
+// renameAndMatchPerms tries 2 ways to rename the file (resorting to a copy if
+// this is across filesystem boundaries), then matches the dest file permissions
+// to the given FileInfo.
+//
+// If source doesn't exist, but dest does, assumes the rename was done
+// previously and just tries to match the permissions.
+func (t *Tidy) renameAndMatchPerms(source, dest string) error {
+	if _, err := os.Stat(source); errors.Is(err, os.ErrNotExist) {
+		if _, err = os.Stat(dest); err == nil {
+			return t.PermsOK(dest)
+		}
+	}
+
+	err := os.Rename(source, dest)
+	if err != nil {
+		if err = shutil.CopyFile(source, dest, false); err != nil {
+			return err
+		}
+	}
+
+	return t.PermsOK(dest)
+}
+
+// PermsOK checks whether the given file has the same ownership and read-write
+// permissions as our destDir.
+func (t *Tidy) PermsOK(path string) error {
+	current, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if err = ownershipMatches(path, current, t.DestDirInfo); err != nil {
+		return err
+	}
+
+	return t.readWriteMatches(path, current)
+}
+
+// ownershipMatches checks whether the given file with the current fileinfo
+// has the same user and group ownership as the desired fileinfo.
+func ownershipMatches(path string, current, desired fs.FileInfo) error {
+	uid, gid := getUIDAndGID(current)
+	desiredUID, desiredGID := getUIDAndGID(desired)
+
+	if uid == desiredUID && gid == desiredGID {
+		return nil
+	}
+
+	return os.Lchown(path, desiredUID, desiredGID)
+}
+
+// getUIDAndGID extracts the UID and GID from a FileInfo. NB: this will only
+// work on linux.
+func getUIDAndGID(info fs.FileInfo) (int, int) {
+	return int(info.Sys().(*syscall.Stat_t).Uid), int(info.Sys().(*syscall.Stat_t).Gid) //nolint:forcetypeassert
+}
+
+// readWriteMatches ensures that the given file with the current fileinfo has the
+// same user,group,other read&write permissions as our destDir.
+func (t *Tidy) readWriteMatches(path string, current fs.FileInfo) error {
+	currentMode := current.Mode()
+	currentRW := currentMode & modeRW
+	desiredRW := t.DestDirInfo.Mode() & modeRW
+
+	if currentRW == desiredRW {
+		return nil
+	}
+
+	return os.Chmod(path, currentMode|desiredRW)
+}
+
+// moveBaseDirsFile moves the base.dirs file in sourceDir to a uniquely named
+// .basedirs file in destDir that includes our date.
+func (t *Tidy) moveBaseDirsFile(inSuffix, outSuffix string) error {
+	source := filepath.Join(t.SrcDir, inSuffix)
+
+	dest := filepath.Join(t.DestDir, fmt.Sprintf("%s_%s.%s",
+		t.Date,
+		filepath.Base(t.SrcDir),
+		outSuffix))
+
+	return t.renameAndMatchPerms(source, dest)
+}
+
+// findAndMoveDBs finds the combine.dgut.db directories in our sourceDir
+// and moves them to a uniquely named dir in destDir that includes our
```

Comment:
I don't think all paragraphs of all help texts have been properly rewrapped. Go through everything to fix.